Rename CRT to Threaded Discussions

marianunez opened the pull request:
#### Summary

Rename Collapsed Reply Threads to Threaded Discussions in the Admin Console screens. No changes are done to config settings to avoid migration changes. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57431

#### Screenshots

|  before  |  after  |
|----|----|
| <img width="1088" alt="Screenshot 2024-07-25 at 1 54 20 PM" src="https://github.com/user-attachments/assets/d6297729-78a1-4583-80bc-190f12e11849"> | <img width="999" alt="Screenshot 2024-07-25 at 2 15 48 PM" src="https://github.com/user-attachments/assets/ccb4109d-ca69-43bd-b42b-1b48cd86cdfc"> |

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates).
* API additions—new endpoint, new response fields, or newly accepted request parameters.
* Database changes (any).
* Schema migration changes. Use the [Schema Migration Template](https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?usp=sharing) as a starting point to capture these details as release notes. 
* Websocket additions or changes.
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating).
* New features and improvements, including behavioral changes, UI changes, and CLI changes.
* Bug fixes and fixes of previous known issues.
* Deprecation warnings, breaking changes, or compatibility notes.

If no release notes are required, write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```
NONE
```
-->
```release-note
Rename Collapsed Reply Threads to Threaded Discussions in System Console
```

M-ZubairAhmed reviewed (changes requested):
We have the term CRT even in the user settings as well. Dont we want to change there as well?

cwarnermm commented:
I agree with @M-ZubairAhmed that we have CRT references in user preferences that also need to be renamed. 

There's also the 2 modal popups about CRT threads (one for admins and one for end users); not sure how/when the end user one is triggered, but worth reviewing/updating.

![Screenshot 2024-07-26 at 8 43 40 AM](https://github.com/user-attachments/assets/e718afa1-8f55-44f7-8ec2-dd0b810ea31d)

![Screenshot 2024-07-26 at 8 42 59 AM](https://github.com/user-attachments/assets/58cab9c7-a148-409c-8099-28571198183a)

marianunez commented:
> We have the term CRT even in the user settings as well. Dont we want to change there as well?

Yes, I had changed the title in the translation file for the user settings. Just missed the Default Message. Added

marianunez commented on webapp/channels/src/components/admin_console/admin_definition.tsx:
```diff
@@ -2515,16 +2515,16 @@ const AdminDefinition: AdminDefinitionType = {
-                            label: defineMessage({id: 'admin.experimental.collapsedThreads.title', defaultMessage: 'Collapsed Reply Threads'}),
-                            help_text: defineMessage({id: 'admin.experimental.collapsedThreads.desc', defaultMessage: 'When enabled (default off), users must enable collapsed reply threads in Settings. When disabled, users cannot access Collapsed Reply Threads. Please review our <linkKnownIssues>documentation for known issues</linkKnownIssues> and help provide feedback in our <linkCommunityChanne ...
+                            label: defineMessage({id: 'admin.experimental.collapsedThreads.title', defaultMessage: 'Threaded Discussions'}),
+                            help_text: defineMessage({id: 'admin.experimental.collapsedThreads.desc', defaultMessage: 'When enabled (default off), users must enable Threaded Discussions in Settings. When disabled, users cannot access Threaded Discussions. Please review our <linkKnownIssues>documentation for known issues</linkKnownIssues> and help provide feedback in our <linkCommunityChannel>Comm ...
```
@esethna do we still add this link to known issues for CRT in the System Console now that is no longer in Beta?

marianunez commented:
> There's also the 2 modal feature discovery popups about CRT threads (one for admins and one for end users); not sure how/when the end user one is triggered, but worth reviewing/updating.

@cwarnermm those modals were product notices that we push out from a different repository and are not part of the product. Those are no longer live so no additional changes required for them.

cwarnermm commented:
@marianunez - I am able to make the admin modal show up in my cloud instance by setting CRT to off by default. Would that one be part of the product and something to update?

marianunez commented:
/update-branch

esethna commented:
Also realized, there is a CRT end user tutorial tour for first time users. Not sure if that cas any references to "Collapsed Reply Threads"? @marianunez

amyblais commented:
> can you help update the term "Collapsed Reply Threads" > "Threaded Discussions" in any notices we are currently shipping?

Opened https://github.com/mattermost/notices/pull/410.

marianunez commented:
> Also realized, there is a CRT end user tutorial tour for first time users. Not sure if that has any references to "Collapsed Reply Threads"? @marianunez

@esethna Double checked and none of the wording in the Tutorial mentions the feature `Collapsed Reply Threads or CRT`. The tutorial references the new areas of `Threads`, `Follow`, `Unreads`, etc.

M-ZubairAhmed reviewed (approved):
Please also cross check the E2Es as well

marianunez commented:
> Please also cross check the E2Es as well

Double checked the E2E tests and there is no action to change CRT settings or any action that uses the UI elements. All tests are using the config directly that was not impacted with this change. 

Thanks @M-ZubairAhmed!

amyblais commented:
@cwarnermm Kind reminder for this PR review.

mattermost-build commented:
Will try to auto merge this PR once all tests and checks are passing. This might take up to an hour.

mattermost-build commented:
Trying to auto merge this PR.

mattermost-build commented:
Pull Request successfully merged
SHA: 988939521eeb5a356ed19e558fba43c7ded04e6b
